BaseClasses.py

```diff
@@ -542,7 +542,9 @@ def has_beaten_game(self, state: CollectionState, player: Optional[int] = None)
         else:
             return all((self.has_beaten_game(state, p) for p in range(1, self.players + 1)))
 
-    def can_beat_game(self, starting_state: Optional[CollectionState] = None) -> bool:
+    def can_beat_game(self,
+                      starting_state: Optional[CollectionState] = None,
+                      locations: Optional[Iterable[Location]] = None) -> bool:
```

Comment:
```suggestion
                      locations: Optional[Iterable[Location]] = None) -> bool:
        """
        If `locations` is given, this will return whether we can beat the game
        with only the locations given (rather than allowing all the locations
        in the multiworld if `locations` is `None`).
        """
```
It was not clear at first glance what the `locations` parameter would do.

Reply:
Hmmm. I see what you're saying but I'm not immediately 100% on it. Multiple core functions allow passing a list of locations, and it always means the same thing: Only consider these locations when sweeping/whatever

Basically I think I'd have the same criticism of functions such as `sweep_for_advancements` or `sweep_from_pool` then.

Which doesn't mean I can't do it here and set a precedent though, so I'll definitely consider it